apps/examples/nettest/nettest.c

```diff
@@ -307,24 +392,26 @@ void ipmcast_receiver_thread(int num_packets)
 		ret = read(sd, databuf, datalen);
 
 		if (ret < 0) {
-			printf("\n[MCASTSERV] ERR: reading datagram message\n");
-		} else {
-			databuf[ret] = '\0';
-			count++;
-			if (count > num_packets) {
-				printf("\n[MCASTSERV] - Received Msg # %d] read (%s) (%d) bytes, Terminating MCLIENT as received sufficient packets for testing\n", count, databuf, ret);
-				break;
-			} else {
-				printf("\n[MCASTSERV] - Received Msg # %d] read (%s) (%d) bytes\n", count, databuf, ret);
-			}
+			printf("[MCASTSERV] ERR: reading datagram message (%d)\n", errno);
+			break;
+		}
+
+		count++;
+		printf("[MCASTSERV] - Received Msg # %d] read (%d) bytes\n", count, ret);
+		if (num_packets == 0) {
+			continue;
+		}
+		if (count > num_packets) {
+			printf("[MCASTSERV] Terminating MCLIENT as received sufficient packets for testing\n");
+			break;
 		}
 	}
 	close(sd);
 	return;
 
 errout_with_socket:
```

Comment:
Please, remove two lines (413~414) and move 'errout_with_socket' between line408 and line 409.